### PR TITLE
Don't modify the headers object that is passed in.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/connection.toit [hunk 1]
```diff
@@ -30,24 +31,33 @@ class Connection:
...
         body_writer = ChunkedWriter writer_
     else:
+      // Return a writer that doesn't accept any data.
       body_writer = ContentLengthWriter this writer_ 0
 
     socket_.set_no_delay false
 
     writer_.write status
     headers.write_to writer_
+    if needs_to_write_chunked_header:
```
Comment (kasperl) [inline src/connection.toit:57]: Nit: I think I'd prefer having Host: before this.

File: src/headers.toit [hunk 2]
```diff
@@ -77,12 +85,22 @@ class Headers:
...
+      result.headers_[key] = copied
+    return result
+
   // Camel-case a string.  Only works for ASCII in accordance with the HTTP
   // standard.  If the string is already camel cased (the norm) then no
   // allocation occurs.
   ascii_normalize_ str:
     alpha := false  // Was the previous character an alphabetic (ASCII) letter.
-    ba := null  // Allocate byte array later if needed.
+    ba /ByteArray? := null  // Allocate byte array later if needed.
```
Comment (kasperl) [inline src/headers.toit:103]: I would call this buffer or something else. Calling it `ba` smells too much like Go to me.

Also, I'd remove the space before `/`.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/connection.toit (hunk 1 above)
Comment (floitsch) [reply src/connection.toit]: done.

File: src/headers.toit (hunk 2 above)
Comment (floitsch) [reply src/headers.toit]: done.

**Pull request merged**